src/sql-parser/src/parser.rs

```diff
@@ -2057,6 +2081,42 @@ impl<'a> Parser<'a> {
         })
     }
 
+    fn parse_avro_doc_on_option_name(&mut self) -> Result<DocOnIdentifier<Raw>, ParserError> {
+        match self.expect_one_of_keywords(&[TYPE, COLUMN])? {
+            TYPE => Ok(DocOnIdentifier::Type(self.parse_raw_name()?)),
+            COLUMN => {
+                let (item_name, column_name) = self.parse_column_name()?;
+                Ok(DocOnIdentifier::Column {
+                    item_name,
+                    column_name,
+                })
+            }
+            _ => unreachable!(),
+        }
+    }
+
+    fn parse_column_name(&mut self) -> Result<(RawItemName, Ident), ParserError> {
+        let start = self.peek_pos();
+        let mut item_name = self.parse_raw_name()?;
```

Comment:
This differs from https://github.com/MaterializeInc/materialize/blob/8b47d4c7df7deeb6fe8645262f59fb4a7d2ce60f/src/sql-parser/src/parser.rs#L7764 which calls parse_identifiers. I thiiiink the one in this PR is correct because that's what AstInfo expects. @ParkMyCar fyi: comment on column should probably be taught to call this new function once it supports AstInfo.